infra/azure-peerpods-iam/main.tf

```diff
@@ -0,0 +1,75 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "4.5.0"
+    }
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = "3.0.2"
+    }
+    local = {
+      source  = "hashicorp/local"
+      version = "2.5.2"
+    }
+  }
+}
+
+provider "azurerm" {
+  subscription_id = var.subscription_id
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+}
+
+data "azurerm_subscription" "current" {}
+
+data "azuread_client_config" "current" {}
+
+provider "azuread" {
+  tenant_id = data.azurerm_subscription.current.tenant_id
+}
+
+locals {
+  name = var.resource_group
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = var.resource_group
```

Comment:
We should still add a suffix somewhere to make this azure/peerpod specific. Otherwise rg might collide with aks/nested setup.

Also, upload-image step still expects a the _caa_cluster suffix.

Reply:
Ok, I added the `_caa_cluster` suffix back. Original reason for using the literal group was so that I can reuse `contrast-ci`, but now that the RG is owned by `create-pre` it does not matter anymore.